src/resources/filters/common/debug.lua

```diff
@@ -44,4 +36,88 @@ function tdump (tbl, indent, refs)
   end
 end
 
+-- "asLua" knows about pandoc opaque objects and converts them to Lua
+-- objects ahead of debug printing
+function asLua(o)
+  if type(o) ~= 'userdata' then
+    return o
+  end
+  local p = string.format("%p", o)
 
+  -- we check equality against the addresses to work around
+  -- https://github.com/jgm/pandoc/issues/8267
+  if p == string.format("%p", PANDOC_READER_OPTIONS) then
```

Comment:
This is a beautiful hack. The problem in 8267 is the `__eq` metamethod, which we could skip with `rawequal(o, PANDOC_READER_OPTIONS)`.

Reply:
TIL about `rawequal`, thank you!